lib/viewAction.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { Node } from './files/node.ts'
+import { View } from './navigation/view.ts'
+
+export interface ViewActionData {
+	/** Unique ID */
+	id: string
+
+	/** Translated name of the action */
+	displayName: string
+
+	/** Translated title of the action */
+	title?: string
+
+	/** Raw svg string */
+	iconSvgInline: string
```

Comment:
There is no other icon, so maybe make the name shorter and more succinct and name it `icon`?

Reply:
I actually prefer to stay consistent between actions @susnux

Reply:
I have not checked the other but of they also use that name then yes :)